File: src/main/java/me/kuku/yuq/logic/LeXinMotionLogic.java
package me.kuku.yuq.logic;

import com.IceCreamQAQ.Yu.annotation.AutoBind;
import me.kuku.yuq.entity.MotionEntity;
import me.kuku.yuq.pojo.Result;

import java.io.IOException;
import java.io.InputStream;

@AutoBind
public interface LeXinMotionLogic {
    InputStream getCaptchaImage(String phone) throws IOException;
    Result<String> getCaptchaCode(String phone, String captchaImageCode) throws IOException;
    Result<MotionEntity> loginByPassword(String phone, String password) throws IOException;
    Result<MotionEntity> loginByPhoneCaptcha(String phone, String captchaPhoneCode) throws IOException;
    String modifyStepCount(int step, MotionEntity motionEntity) throws IOException;
    String bindBand(MotionEntity motionEntity) throws IOException;
}

File: src/main/kotlin/me/kuku/yuq/logic/LeXinMotionLogic.kt
package me.kuku.yuq.logic

import com.IceCreamQAQ.Yu.annotation.AutoBind
import me.kuku.yuq.entity.MotionEntity
import me.kuku.yuq.entity.QQLoginEntity
import me.kuku.yuq.pojo.CommonResult

@AutoBind
interface LeXinMotionLogic {

    fun getCaptchaImage(phone: String): ByteArray

    fun getCaptchaCode(phone: String, captchaImageCode: String): CommonResult<String>

    fun loginByPassword(phone: String, password: String): CommonResult<MotionEntity>

    fun loginByPhoneCaptcha(phone: String, captchaPhoneCode: String): CommonResult<MotionEntity>

    fun loginByQQ(qqLoginEntity: QQLoginEntity): CommonResult<MotionEntity>

    fun modifyStepCount(step: Int, motionEntity: MotionEntity): String

    fun bindBand(motionEntity: MotionEntity): String

}